src/Symfony/Component/Messenger/Bridge/Kafka/Transport/KafkaSender.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Messenger\Bridge\Kafka\Transport;
+
+use Psr\Log\LoggerInterface;
+use RdKafka\Producer as KafkaProducer;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Exception\TransportException;
+use Symfony\Component\Messenger\Transport\Sender\SenderInterface;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+
+/**
+ * @author Konstantin Scheumann <konstantin@konstantin.codes>
+ */
+class KafkaSender implements SenderInterface
+{
+    /** @var LoggerInterface */
+    private $logger;
+
+    /** @var SerializerInterface */
+    private $serializer;
+
+    /** @var RdKafkaFactory */
+    private $rdKafkaFactory;
+
+    /** @var KafkaSenderProperties */
+    private $properties;
+
+    /** @var KafkaProducer */
+    private $producer;
+
+    public function __construct(
+        LoggerInterface $logger,
+        SerializerInterface $serializer,
+        RdKafkaFactory $rdKafkaFactory,
+        KafkaSenderProperties $properties
+    ) {
+        $this->logger = $logger;
+        $this->serializer = $serializer;
+        $this->rdKafkaFactory = $rdKafkaFactory;
+        $this->properties = $properties;
+    }
+
+    public function send(Envelope $envelope): Envelope
+    {
+        $producer = $this->getProducer();
+        $topic = $producer->newTopic($this->properties->getTopicName());
+
+        $payload = $this->serializer->encode($envelope);
+
+        $topic->producev(
+            \RD_KAFKA_PARTITION_UA,
+            0,
+            $payload['body'],
+            $payload['key'] ?? null,
```

Comment:
Use stamps if you want to provide additional information to the envelope (see SQS and Fifo for instance)